### PR TITLE
D1 Migrations

### PR DESCRIPTION
**Pull request by G4brym**
This adds `wrangler d1 migrations` sub command, this is still a WIP

```
wrangler d1 migrations

Interact with D1 Migrations

Commands:
  wrangler d1 migrations list <name>    List your D1 migrations
  wrangler d1 migrations create <name> <message>  Create a new Migration
  wrangler d1 migrations apply <name>   Apply D1 Migrations

Flags:
  -c, --config   Path to .toml configuration file  [string]
  -h, --help     Show help  [boolean]
  -v, --version  Show version number  [boolean]
```

### CODE REVIEW

File: packages/wrangler/src/d1/migrations.tsx [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+								name       TEXT UNIQUE,
+								applied_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+						);
+				`
+	);
+};
+
+type MigrationsCreateArgs = { config?: string; name: string; message: string };
+
+export function CreateOptions(yargs: Argv): Argv<MigrationsCreateArgs> {
```
Comment (WalshyDev) [inline packages/wrangler/src/d1/migrations.tsx:335]: why is this so far above the handler? can we keep them together

File: packages/wrangler/src/d1/index.ts [hunk 2]
```diff
@@ -70,6 +71,28 @@ export const d1 = (yargs: Argv) => {
...
+						Migrations.ListHandler
+					)
+					.command(
+						"create <name> <message>",
+						"Create a new Migration",
+						Migrations.CreateOptions,
+						Migrations.CreateHandler
+					)
+					.command(
+						"apply <name>",
```
Comment (WalshyDev) [inline packages/wrangler/src/d1/index.ts:89]: I feel like name is confusing here. It made me check the code to see if this only applied one migration. We should really have as useful naming as possible

```suggestion
						"list <database>",
						"List your D1 migrations",
						Migrations.ListOptions,
						Migrations.ListHandler
					)
					.command(
						"create <database> <message>",
						"Create a new Migration",
						Migrations.CreateOptions,
						Migrations.CreateHandler
					)
					.command(
						"apply <database>",
```

File: packages/wrangler/src/d1/migrations.tsx (hunk 1 above)
Comment (G4brym) [reply packages/wrangler/src/d1/migrations.tsx]: Fixed

File: packages/wrangler/src/d1/index.ts (hunk 2 above)
Comment (G4brym) [reply packages/wrangler/src/d1/index.ts]: Agreed, fixed

**Pull request merged**